tests: rkt-monitor add new statistics

borqosky opened the pull request:
New statistics for load average and boot time measure. Godeps updated
as well

steveej committed:
tests/net: add more port forwarding cases

* bridged network
* different host port than application port

steveej committed:
net/portfwd: choose network according to ipmasq

When a network has ipmasq enabled, CNI generates masquerading rules for
the networks subnet. These rules are appended, so the first network is
the most upper one. Hence, rkt needs to use the address of first
masqueraded network as packets will always hit this subnet's CNI rule
first.

If no network has ipmasq enabled, we default to the previous behavior of
using the last loaded network, which is either the default or
default-restricted network.

steveej committed:
net/portfwd: allow host/pod port to differ

tmrts committed:
rkt: simplify if-else chains and spacing

Refactors the if-else chains for readability

Fixes #2421

Commit:
dist: Update tmpfiles to create /etc/rkt

By creating this directory, users can run `rkt trust` without being
root, if the user is in the rkt group.

tmrts committed:
tests: invoke gofmt with simplify-code flag

Enables code simplification checks of gofmt.

Related to #2462

steveej committed:
net/portfwd: refactor related methods

Commit:
api: Return the pods even when we failed to get every information of it.

Commit:
tests: rkt-monitor add new statistics

New statistics for load average and boot/stop time measure of container.
Diferences are measured in a nanoseconds. Godeps updated as well.
Change network from 'host' to 'default-restricted' since kvm flavor
can't create this type.

jellonek commented on tests/rkt-monitor/main.go:
```diff
@@ -106,10 +107,13 @@ func runRktMonitor(cmd *cobra.Command, args []string) {
 	}
 	err = execCmd.Start()
+
+	startTime := time.Now()
```
This line should be probably above above ;)

jellonek commented:
Quite simple and nice. One minor flaw.

jellonek commented:
LGTM

chrskhl committed:
docs: remove indef articles from start of arg doc strings

chrskhl committed:
docs: fix rkt run-prepared's --interactive doc string

pskrzyns commented:
+1

Commit:
test: Update rkt api service tests.

nhlfr committed:
lkvm: Bump the version of upstream repository

glibc >= 2.23 deprecated the readdir_r usage in favor
of readdir. Build of lkvm is treating warnings as errors,
so it's impossible to build rkt with kvm flawor with the
newest glibc.

That was fixed in the upstream commit[1].

Also, the another commit[2] fixed an issue with dead lock
on reboot, so dead_lock_fix patch isn't needed anymore.

[1] https://kernel.googlesource.com/pub/scm/linux/kernel/git/will/kvmtool/+/d62653e177597251c24494a6dda60acd6d846671
[2] https://kernel.googlesource.com/pub/scm/linux/kernel/git/will/kvmtool/+/e8cb90fb9697da24cdf553a1b8cf120f94c20832

Fixes #2594

Signed-off-by: Michal Rostecki <michal.rostecki@gmail.com>

tmrts committed:
stage0: check and create /etc

Checks '/etc' before writing to '/etc/rkt-resolv.conf'
and create it with default permissions if it doesn't exist.

iaguis committed:
stage1/init: create a new mount ns for each app

Up to this point, you could escape the app's chroot easily by using a
simple program downloaded from the internet [[1]][1]. To avoid this, we
now create a new mount namespace per each app.

You can still escape the chroot if you have CAP_SYS_PTRACE and access
`/proc/1/root` but this fixes the issue for images that don't need this
capability.

[1]: http://www.unixwiz.net/techtips/chroot-practices.html

tmrts committed:
Merge pull request #2599 from tmrts/check/etc-exists

stage0: check and create /etc

Fixes #2141

jonboulle committed:
Merge pull request #2595 from nhlfr/glibc-readdir

lkvm: Bump the version of upstream repository

matthaias committed:
docs: Change getting started guide to build a static binary

Current build command would not work after #2506

Fixes #2584

Commit:
Merge pull request #2387 from steveeJ/rkt-bridge-net-tests

net: fix port forwarding behavior with custom CNI ipMasq'ed networks different and allow different hostPort:podPort combinations

jonboulle committed:
Merge pull request #2605 from matthaias/fix-getting-started-guide-with-static-build

docs: Change getting started guide to build a static binary

Commit:
build: check for openssl headers for kvm flavor

To use kernel sources during kvm flavor build, openssl headers
are needed.

Also update information about dependencies in documentation.

euank committed:
stage1: Fix segfault in enterexec

Prior to this commit, if `rkt enter` was executed without the `TERM`
environment variable set, a segfault could occur.
This commit treats an empty TERM environment variable as meaning 'xterm'
and avoids the segfault.

tmrts committed:
tests: simplify for loop conditions

For loop invocation is simplified to comply with `gofmt -s`

matthaias committed:
docs: Add example to run multiple apps in one pod (#2609)

* docs: Add example to run multiple apps in one pod

A example was missing how to run multiple apps in one pod

Fixes #2583

* docs: Change level of "run multiple apps in same pod", add preamble

"run multiple apps in same pod" should be higher level

Fixes #2583

tmrts committed:
Merge pull request #2489 from tmrts/improve/gofmt-check

Invoke gofmt with simplify-code flag

iaguis committed:
functional tests: test reading a file written in another app

alban committed:
Bump Godeps to appc/spec 0.8.0

alban committed:
rkt: fix compilation with appc/spec 0.8.0

The fixes are necessary since the introduction of the following changes:
- https://github.com/appc/spec/pull/561
- https://github.com/appc/spec/pull/576

alban committed:
tests: ACE: fix compilation with appc/spec 0.8.0

The fixes are necessary since the introduction of the following changes:
- https://github.com/appc/spec/pull/569

alban committed:
tests: print proper output for TestAceValidator

This allows to debug https://github.com/coreos/rkt/issues/2171 properly.

matthaias committed:
scripts: add libtspi-dev

Now the install script can be reused e.g. for the 'build with docker' part

Fixes #2215

matthaias committed:
docs: change docker img to debian:sid, install deps via script

"build with docker" and "build with rkt" should use same deps and img

Fixes #2215

matthaias committed:
docs: fixed build with docker cmd (forgot && between two cmds)

Without "&&" the cmd is not correct

Fixes #2215

tmrts committed:
docs: add instructions for nixos (#2614)

Adds instructions for installing rkt on NixOS.

Related to #2202

jonboulle committed:
Merge pull request #2608 from euank/stage1-enter-segfault

stage1: Fix segfault in enterexec

jonboulle committed:
Merge pull request #2615 from matthaias/add-libtspi-dev-to-install-script-update-docker-build-cmd

build: add libtspi dev to debian install script, update docker build cmd

jonboulle committed:
Merge pull request #2593 from yifan-gu/return_partial_pod

api: Return the pods even when we failed to get every information of it.

iaguis committed:
Merge pull request #2453 from tmrts/simplify/calculateDataDir

Simplify if-else chains and spacing

s-urbaniak committed:
godeps: update cni to v0.2.3

Fixes #2580

iaguis committed:
fuctional tests: check that each app is in a different mount ns

alban committed:
metadata-service: ACE: fix compilation with appc/spec 0.8.0

The fixes are necessary since the introduction of the following changes:
- https://github.com/appc/spec/pull/451

tmrts committed:
Merge pull request #2618 from s-urbaniak/bump-cni

godeps: update cni to v0.2.3

iaguis committed:
Merge pull request #2603 from kinvolk/iaguis/mount-ns

stage1/init: create a new mount ns for each app

jonboulle committed:
Merge pull request #2472 from chancez/create_etc_rkt_tmpfiles

dist: Update tmpfiles to create /etc/rkt

tmrts committed:
Merge pull request #2613 from kinvolk/alban/debug-TestAceValidator

tests: print proper output for TestAceValidator

iaguis committed:
functional tests: ignore ACE validator test

After https://github.com/appc/spec/pull/451, the spec changed regarding
the metadata service.

Since its use is not widespread, let's disable the test for now and fix
it later.

jonboulle committed:
Merge pull request #2588 from kinvolk/ckuehl/simple-arg-doc-string-fixes

stage0: simple arg doc string fixes

alban committed:
Merge pull request #2611 from kinvolk/alban/bump-appc-spec-0.8.0

Bump Godeps to appc/spec 0.8.1

Commit:
tests: save interval data measures to csv file

Save data to csv - useful for plot a graph of usage.
Add a flag command for additional validation
